pkg/share/manager/json/json.go

```diff
@@ -154,6 +191,31 @@ func genID() string {
 	return uuid.New().String()
 }
 
+func (m *mgr) reload() error {
+	fmt.Printf("json.go: reloading file \"%s\"", m.model.file)
+	// TODO: recreate the file based on share data from the storage
+	model, err := loadOrCreate(m.model.file)
+	if err != nil {
+		return errors.Wrap(err, "error loading the file containing the shares")
+	}
+	m.model = model
+	return nil
+}
+
+func (m *mgr) Lock() {
+	m.locked = true
```

Comment:
maybe instead of overriding Lock/Unlock, could simply set the flag in the Save() method before and after writing to catch the events that were triggered by us